### PR TITLE
Feature repository pattern

### PR DESCRIPTION
**Pull request by Alexbeis**
Caso de Uso: Eliminar un video.

### CODE REVIEW

File: src/Mooc/Videos/Application/Remove/VideoRemover.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+	public function __invoke(VideoId $id): void
+	{
+		$video = $this->repository->search($id);
+
+		if (null !== $video) {
+			$this->repository->save($video);
+		} else {
+			throw new VideoNotFound($id);
+		}
```
Comment (adrG2) [inline src/Mooc/Videos/Application/Remove/VideoRemover.php]: Clausula de guarda:
if($video === null) {
	throw new VideoNotFound($id);
}
$this->repository->save($video);

Aunque ya de por sí esa función hace más de una cosa. Yo extraería la lógica de la búsqueda y el tratamiento de errores fuera.

File: .gitignore [hunk 2]
```diff
@@ -14,3 +14,5 @@ vendor/
 .env
 
 build/
+
+.idea
```
Comment (rgomezcasas) [inline .gitignore]: 👉👈I know you already know about it, buuuuuut....

🔥https://pro.codely.tv/library/git-introduccion-y-trabajo-en-equipo/93728/path/step/59548217/ 🔥

File: applications/mooc_backend/src/Controller/Video/VideoRemoveController.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+use CodelyTv\Shared\Infrastructure\Api\Controller\ApiController;
+use CodelyTv\Shared\Infrastructure\Api\Response\ApiHttpCreatedResponse;
+use Symfony\Component\HttpFoundation\Request;
+
+class VideoRemoveController extends ApiController
+{
+
+	protected function exceptions(): array
+	{
+
```
Comment (rgomezcasas) [inline applications/mooc_backend/src/Controller/Video/VideoRemoveController.php]: return empty array here if we're not raising any exception

File: applications/mooc_backend/src/Controller/Video/VideoRemoveController.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+			new Uuid($request->get('request_id')),
+			$request->get('id')
+		);
+
+		$this->dispatch($command);
+
+		return new ApiHttpCreatedResponse();
+
+	}
+}
```
Comment (rgomezcasas) [inline applications/mooc_backend/src/Controller/Video/VideoRemoveController.php]: There are a lot of newlines here 😂

File: applications/mooc_backend/src/Controller/Video/VideoRemoveController.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+	{
+
+		$command = new RemoveVideoCommand(
+			new Uuid($request->get('request_id')),
+			$request->get('id')
+		);
+
+		$this->dispatch($command);
+
+		return new ApiHttpCreatedResponse();
```
Comment (rgomezcasas) [inline applications/mooc_backend/src/Controller/Video/VideoRemoveController.php]: When doing deletes the most common response is `no content` https://restfulapi.net/http-status-204-no-content/ 😊

File: src/Mooc/Videos/Application/Remove/RemoveVideoCommandHandler.php [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+		$this->remover = $remover;
+	}
+
+	public function __invoke(RemoveVideoCommand $command)
+	{
+		$id = new VideoId($command->id());
+
+		$this->remover->remove($id);
+	}
+}
```
Comment (rgomezcasas) [inline src/Mooc/Videos/Application/Remove/RemoveVideoCommandHandler.php]: Maybe we could publish some domain event here? :P

File: src/Mooc/Videos/Application/Remove/VideoRemover.php [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+	public function remove(VideoId $id): void
+	{
+		$video = $this->repository->search($id);
+
+		if (null !== $video) {
+			$this->repository->remove($video);
+		} else {
+			throw new VideoNotFound($id);
+		}
```
Comment (rgomezcasas) [inline src/Mooc/Videos/Application/Remove/VideoRemover.php]: We could do a guard clause for this

**Comment by rgomezcasas** [comment]
Hey!

Since you've got your feedback I'm closing this pull request.

Feel free to comment if you have some doubts!

**Pull request closed without merging**